packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -29,350 +30,285 @@ import { createRef } from "./shared";
+  deduplicatedBundle.total = deduplicatedBundle.entry.length;
 
   return deduplicatedBundle;
 }
 
-/**
- * Finds and updates references to the deduplicated resources
- */
-function replaceResourceReferences(
-  resourceArrays: ExtractedFhirTypes,
-  idMap: Map<string, string[]>
-): ExtractedFhirTypes {
-  let updatedArrays = JSON.stringify(resourceArrays);
-  for (const [masterRef, consumedRefs] of idMap.entries()) {
-    for (const consumedRef of consumedRefs) {
-      const regex = new RegExp(consumedRef, "g");
-      updatedArrays = updatedArrays.replace(regex, masterRef);
-    }
-  }
-
-  return JSON.parse(updatedArrays);
-}
-
-type ResourceFilter = (entry: Resource, link: string) => Resource | undefined;
-
-const allergiesFiltersMap = new Map<string, ResourceFilter>([
-  ["Practitioner", removeDanglingReferences],
-]);
-
-const compositionFiltersMap = new Map<
-  string,
-  typeof removeResource | typeof removeDanglingReferences
->([["all", removeDanglingReferences]]);
-
-const medicationRelatedFiltersMap = new Map<string, ResourceFilter>([
-  ["Medication", removeResource],
-]);
-
-const encounterFiltersMap = new Map<string, ResourceFilter>([
-  ["Condition", removeDanglingReferences],
-  ["Location", removeDanglingReferences],
-  ["Practitioner", removeDanglingReferences],
-]);
-
-const conditionsFiltersMap = new Map<string, ResourceFilter>([
-  ["Practitioner", removeDanglingReferences],
-]);
-
-const coveragesFiltersMap = new Map<string, ResourceFilter>([
-  ["Organization", removeDanglingReferences],
-]);
-
-const diagReportFiltersMap = new Map<string, ResourceFilter>([
-  ["Observation", removeDanglingReferences],
-  ["Encounter", removeDanglingReferences],
-  ["Practitioner", removeDanglingReferences],
-  ["Organization", removeDanglingReferences],
-]);
-
-const observationFiltersMap = new Map<string, ResourceFilter>([
-  ["Practitioner", removeDanglingReferences],
-  ["Organization", removeDanglingReferences],
-]);
-
-const procedureFiltersMap = new Map<string, ResourceFilter>([
-  ["Practitioner", removeDanglingReferences],
-  ["Organization", removeDanglingReferences],
-]);
-
-const immunizationFiltersMap = new Map<string, ResourceFilter>([
-  ["Practitioner", removeDanglingReferences],
-  ["Organization", removeDanglingReferences],
-]);
-
-const resourceFiltersMap = new Map<string, Map<string, ResourceFilter>>([
-  ["AllergyIntolerance", allergiesFiltersMap],
-  ["Condition", conditionsFiltersMap],
-  ["Coverage", coveragesFiltersMap],
-  ["DiagnosticReport", diagReportFiltersMap],
-  ["Encounter", encounterFiltersMap],
-  ["Immunization", immunizationFiltersMap],
-  ["MedicationStatement", medicationRelatedFiltersMap],
-  ["MedicationRequest", medicationRelatedFiltersMap],
-  ["MedicationAdministration", medicationRelatedFiltersMap],
-  ["Composition", compositionFiltersMap],
-  ["Observation", observationFiltersMap],
-  ["Procedure", procedureFiltersMap],
-]);
-
 export function removeResourcesWithDanglingLinks(
```

Comment:
old runtime here was 
O(e **2) since e entries and n links where n approach e at the limit 

new runtime is O(e)